src/mlpack/core/data/load_csv.hpp

```diff
@@ -94,45 +153,58 @@ class LoadCSV
     std::string line;
     while (std::getline(inFile, line))
     {
-      ++rows;
+       ++rows;
+    }
+
+    // Reset the DatasetInfo object, if needed.
+    if (info.Dimensionality() == 0)
+    {
+      info.SetDimensionality(rows);
+    }
+    else if (info.Dimensionality() != rows)
+    {
+      std::ostringstream oss;
+      oss << "data::LoadCSV(): given DatasetInfo has dimensionality "
+          << info.Dimensionality() << ", but data has dimensionality "
+          << rows;
+      throw std::invalid_argument(oss.str());
     }
-    info = DatasetMapper<MapPolicy>(rows);
 
     // Now, jump back to the beginning of the file.
     inFile.clear();
     inFile.seekg(0, std::ios::beg);
     rows = 0;
-
+    
     while (std::getline(inFile, line))
     {
       ++rows;
-      // Remove whitespace from either side.
-      boost::trim(line);
 
       if (rows == 1)
       {
         // Extract the number of columns.
-        auto findColSize = [&cols](iter_type) { ++cols; };
-        qi::parse(line.begin(), line.end(),
-            stringRule[findColSize] % delimiterRule);
+	std::pair<int, int> dimen = GetMatSize(inFile);
+        cols = dimen.second;
       }
 
       // I guess this is technically a second pass, but that's ok... still the
       // same idea...
       if (MapPolicy::NeedsFirstPass)
       {
         // In this case we must pass everything we parse to the MapPolicy.
-        auto firstPassMap = [&](const iter_type& iter)
-        {
-          std::string str(iter.begin(), iter.end());
-          boost::trim(str);
```

Comment:
@rcurtin Do you have an idea why it was removing the white spaces from the string, I think it is removing the trailing white spaces, those at the start and at the end of the string. Otherwise, I fail to see the reason here.

Reply:
I commented `boost::trim(str)` everywhere and ran the old parser.
Nothings seems to change even if we remove it.

Maybe it is to handle malformed csv files? Not sure.

Reply:
What do you mean nothing changes when remove it, if you mean compiling, then use it should compile without any issues.
However, does it pass the tests??

Reply:
I marked the PR as ready for review hoping that it will run the tests, but it's not. I am running the tests locally for now.

How can I run the tests just like before on CI?

Reply:
@heisenbuug there is a conflict in `load_csv.hpp`. Merge the latest master, resolve the conflict and push again. Then all the CI tests will run. :)

Reply:
@RishabhGarg108 I merged the latest master, I don't understand how can there be a conflict here as there is only one branch on which I am working...

Reply:
Are you sure that the merge is successful? I am not sure if it is because generally after merging any branch, you will have a commit message like "Merged branch master of xyz into new-parser".

Also, look below the CI checks in this PR, It says "This branch has conflicts that must be resolved". Then it also mentions the file in which the conflict is `src/mlpack/core/data/load_csv.hpp`.

Reply:
Yea, the conflict has not resolved yet, still working on it.

Reply:
Let me know if the tests are passing locally, there is no need yet to run it on the CI, especially as the pull request is not ready yet.
You need to fetch the `master` branch on you machine and merge with your local master branch.
Once this is done, you can checkout to your branch and type `git merge master`.👍 
At this level, git will tell you that there is a merge conflict in a specific file. You will have to open this file and manage the conflict by hand, e.g. remove the new modification from the master and keep yours.
Once you have resolved all the conflict, you can save the file and add the file using `git add` and them commit
You will have a ready commit message telling that you are merging the master to your branch.
Once this is done you can push the modification and the CI will restart to run.

As said, if the tests are failing locally there is no need to push the modification, as they will fail here too.

Let me know if this is helpful👍